src/common/arrstr.cpp

```diff
@@ -573,3 +579,188 @@ wxArrayString wxSplit(const wxString& str, const wxChar sep, const wxChar escape
+        }
+
+        if ( naturalAlpha.Matches(text) )
+        {
+            naturalAlpha.GetMatch(&alphaStart, &alphaLength, 0);
+        }
+
+
+        if ( alphaStart == 0 )
+        {
+            fragment.text = text.Mid(0, alphaLength);
+            fragment.value = 0;
+            fragment.type = wxFRAGMENT_TYPE_ALPHA;
+
+            text.erase(0, alphaLength);
+        }
+
+        if ( digitStart == 0 )
+        {
+            fragment.text = text.Mid(0, digitLength);
+            fragment.text.ToLong(&fragment.value);
+            fragment.type = wxFRAGMENT_TYPE_DIGIT;
+
+            text.erase(0, digitLength);
+        }
+
+        return fragment;
+    }
+
+    int CompareFragmentNatural(const wxStringFragment& lhs, const wxStringFragment& rhs)
+    {
+        if ( (lhs.type == wxFRAGMENT_TYPE_ALPHA) &&
+             (rhs.type == wxFRAGMENT_TYPE_ALPHA) )
+        {
+            return lhs.text.CmpNoCase(rhs.text);
+        }
+
+        if ( (lhs.type == wxFRAGMENT_TYPE_DIGIT) &&
+             (rhs.type == wxFRAGMENT_TYPE_DIGIT) )
+        {
+            if ( lhs.value == rhs.value )
+            {
+                return  0;
+            }
+
+            if ( lhs.value < rhs.value )
+            {
+                return -1;
+            }
+
+            if ( lhs.value > rhs.value )
+            {
+                return  1;
+            }
+        }
+
+        if ( (lhs.type == wxFRAGMENT_TYPE_DIGIT) &&
+             (rhs.type == wxFRAGMENT_TYPE_ALPHA) )
+        {
+            return -1;
+        }
+
+        if ( (lhs.type == wxFRAGMENT_TYPE_ALPHA) &&
+             (rhs.type == wxFRAGMENT_TYPE_DIGIT) )
+        {
+            return 1;
+        }
+
+        if ( lhs.type == wxFRAGMENT_TYPE_EMPTY )
+        {
+            return -1;
+        }
+
+        if ( rhs.type == wxFRAGMENT_TYPE_EMPTY )
+        {
+            return 1;
+        }
+
+        return 0;
+    }
+
+} // unnamed namespace
+
+
+
+// ----------------------------------------------------------------------------
+// wxCmpNaturalNative
+// ----------------------------------------------------------------------------
+// 
+int wxCMPFUNC_CONV wxCmpNatural(const wxString& s1, const wxString& s2)
+{
+    wxString lhs(s1);
+    wxString rhs(s2);
+
+    int comparison = 0;
+
+    while ( (comparison == 0) && (!lhs.empty() || !rhs.empty()) )
+    {
+        wxStringFragment fragmentL = GetFragment(lhs);
+        wxStringFragment fragmentR = GetFragment(rhs);
+        comparison = CompareFragmentNatural(fragmentL, fragmentR);
+    }
+
+    return comparison;
+}
+
+
+// ----------------------------------------------------------------------------
+// Declaration of StrCmpLogicalW()
+// ----------------------------------------------------------------------------
+// 
+// In some distributions of MinGW32, this function is exported in the library,
+// but not declared in shlwapi.h. Therefore we declare it here.
+#if defined( __MINGW32_TOOLCHAIN__ )
+extern "C"
```

Comment:
Now you have `extern "C"` twice there.

I suggest removing the block declaration (`extern "C" {}`, which is usually used when you need to declare several such functions at once) and keeping the in-line one, which should make the code a bit more compact and readable, i.e., changing
```
extern "C"
{
    extern "C" __declspec(dllimport) int WINAPI StrCmpLogicalW(LPCWSTR psz1, LPCWSTR psz2);
}

```
to
`extern "C" __declspec(dllimport) int WINAPI StrCmpLogicalW(LPCWSTR psz1, LPCWSTR psz2);`